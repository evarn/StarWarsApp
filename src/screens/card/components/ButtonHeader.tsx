import React, {Dispatch} from 'react';
import {StyleSheet, View} from 'react-native';
import {useAppSelector} from '../../../redux/hooks';
import selectCharcterData from '../../../redux/store/characterSelector';
import TitleCardScreen from './TitleCardScreen';
import Strings from './../../../constants/strings';
import AppButton from '../../components/appButton/AppButton';

interface IButtonHeader {
  setModalFilterVisible: Dispatch<React.SetStateAction<boolean>>;
  setIsReset: Dispatch<React.SetStateAction<boolean>>;
}

const ButtonHeader = ({setModalFilterVisible, setIsReset}: IButtonHeader) => {
  const {count, people, filterPeople, isFiltred} =
    useAppSelector(selectCharcterData);

  const dataFlatlist = isFiltred ? filterPeople : people;

  const _ListHeaderComponent = () => {
    return (
      <>{dataFlatlist.length ? <TitleCardScreen count={count} /> : undefined}</>
    );
  };

  return (
    <View style={styles.bntContainer}>
      <AppButton
        title={Strings.BUTTON_HEADER_FILTER}
        onPressButton={() => setModalFilterVisible(true)}
        styleView={styles.btn}
      />
      <AppButton
        title={Strings.BUTTON_HEADER_RESET}
        onPressButton={() => setIsReset(true)}
        styleView={styles.btn}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  bntContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginBottom: 12,
  },
  btn: {
    maxWidth: '40%',
    marginHorizontal: 12,
  },
});

export default ButtonHeader;
