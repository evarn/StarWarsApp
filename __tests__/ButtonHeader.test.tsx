import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import ButtonHeader from '../src/screens/card/components/ButtonHeader';
import Strings from '../src/constants/strings';

describe('ButtonHeader', () => {
  const setModalFilterVisible = {function(): void {}};
  const setIsReset = {function(): void {}};
  it('ButtonHeader snapshoot', () => {
    const {toJSON} = render(
      <ButtonHeader
        setModalFilterVisible={() => setModalFilterVisible}
        setIsReset={() => setIsReset}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('ButtonHeader text', () => {
    const {getByText} = render(
      <ButtonHeader
        setModalFilterVisible={() => setModalFilterVisible}
        setIsReset={() => setIsReset}
      />,
    );
    const filter = Strings.BUTTON_HEADER_FILTER;
    const foundTextFilter = getByText(filter);
    const reset = Strings.BUTTON_HEADER_RESET;
    const foundTextReset = getByText(reset);
    expect(foundTextFilter.props.children).toEqual(filter);
    expect(foundTextReset.props.children).toEqual(reset);
  });
});
