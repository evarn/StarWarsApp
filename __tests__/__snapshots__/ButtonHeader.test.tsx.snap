// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonHeader ButtonHeader snapshoot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "marginBottom": 12,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FFC107",
        "borderRadius": 11,
        "height": 66,
        "justifyContent": "center",
        "marginHorizontal": 12,
        "marginTop": 24,
        "maxWidth": "40%",
        "opacity": 1,
        "width": 231,
      }
    }
  >
    <Text
      style={
        {
          "color": "#212121",
          "fontFamily": "Karla",
          "fontSize": 23,
          "fontStyle": "normal",
          "fontWeight": "700",
        }
      }
    >
      Filter
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#FFC107",
        "borderRadius": 11,
        "height": 66,
        "justifyContent": "center",
        "marginHorizontal": 12,
        "marginTop": 24,
        "maxWidth": "40%",
        "opacity": 1,
        "width": 231,
      }
    }
  >
    <Text
      style={
        {
          "color": "#212121",
          "fontFamily": "Karla",
          "fontSize": 23,
          "fontStyle": "normal",
          "fontWeight": "700",
        }
      }
    >
      Reset
    </Text>
  </View>
</View>
`;
